src/media/media.service.ts

```diff
@@ -45,6 +47,37 @@ export class MediaService {
     return { id, identifier, location };
   }
 
+  public async cleanUpImages(): Promise<CleanUpResult> {
+    /*
+     We only delete images which are older than 1 day and match the conditions
+     to avoid accidentally deleting images that are in the upload process.
+     This does only apply for the tour conditions; images without user are
+     immediately deleted.
+    */
+    const bufferDate = dayjs().subtract(1, 'day').toDate();
+    const dateCondition = LessThan(bufferDate);
+    const imagesToClean = await this.imageRepository.find({
+      where: [{ tourId: null, createdAt: dateCondition }, { userId: null }],
+    });
+    const imageIdentifiers = imagesToClean.map((image) => image.identifier);
+
+    const deletedFromStorage = await this.storageProvider.deleteMany(
+      imageIdentifiers,
+    );
+
+    // Only call delete if we have images to delete, otherwise it throws.
+    let deletedFromDb: DeleteResult;
+    if (imagesToClean.length > 0) {
```

Comment:
Why not move the if statement some lines up and also only execute the storageProvider call?

Reply:
You're right. I changed this in https://github.com/gipfeli-io/gipfeli-api/pull/60 by using a DTO (which is cleaner anyways).